pkg/ingester/limiter.go

```diff
@@ -118,10 +121,13 @@ func (l *Limiter) FormatError(userID string, err error) error {
 func (l *Limiter) formatMaxSeriesPerUserError(userID string) error {
 	globalLimit := l.limits.MaxGlobalSeriesPerUser(userID)
 
-	return errors.New(globalerror.MaxSeriesPerUser.MessageWithPerTenantLimitConfig(
-		fmt.Sprintf("per-user series limit of %d exceeded", globalLimit),
-		validation.MaxSeriesPerUserFlag,
-	))
+	return log.SampledError{
```

Comment:
I'd create a constructor `log.NewSampledError(error, *log.Sampler) error`, because the empty value of `log.SampledError` isn't valid and would panic without a `Sampler`. Having a constructor enforces callers to provide both args (they can still do nil, but then you can blame them).

Reply:
I've added code to allow `nil`, so that users can configure the feature off.
Now I think the constructor wouldn't add much.

Reply:
I added a constructor, as a method on the sampler.